### ISSUE TITLE
a few question by a new user

### ISSUE THREAD

**Issue by ginick** [arkenfox/user.js#1200]
Hello
first of all thanks for all the hard work put into this project;
the user.js file is very well put together,includes good explanations and instructions,and the wiki is very helpful.

i was looking in the "implementation" section and was wondering a few things;

1- as i understand;the "binary check" option(0412)  is related to all the Google safe browsing settings and not to what is mentioned in this article: https://www.ghacks.net/2017/03/31/firefox-to-get-option-to-verify-all-binaries-to-protect-against-compromise/
-about section (0302a)"Auto-installing of Firefox updates is disabled"; how do you think i should set this option under Linux?
i suppose leaving it as it is would not prevent  installing new updates from the repository as they come?
Edit:
i checked my current about:config status for "app.update.auto" and it's true, so i guess I'll do the same under the user.js file?

both of the above were concerns i had related to the Firefox update under Linux.

2- i already tried using this user.js file in a VM and while i managed to make everything work the way i like,i noticed quit a delay with almost every action;like after pressing enter and getting the search result,or going back or forward on web pages etc.
i suppose it might be related to the browser not predicting or pre-fetching data?
-how can i get rid of that delay?

3-last one is about uniqueness and fingerprint;
my current browser settings+OS+combination of add-ons already makes it quit unique,so adding this user.js file wouldn't make it even more?

**Comment by rusty-snake** [comment]
1. If you installed firefox via apt/dnf/pacman/... it is build w/o updated support => this pref does nothing
2. Start looking at the `[SETUP-PERF]` tags.

**Comment by Thorin-Oakenpants** [comment]
1. a)

obviously Safe Browsing can't list everything - most (all?) of the local data is essentially deny lists of known bad actors/URLs. Binaries/executables are checked in real time - there shouldn't be a privacy concern here, but with google you never know. IF google were to be using this information (on what binaries you like to download), AND it was leaked, the ENTIRE internet would go bananas, so I doubt they do it: especially given this is a security feature and compromising it's integrity would be a big no-no. Mozilla also have a contract with google re Safe Browsing and privacy, and breaking that would be not only be financially irresponsible, it would also be devastatingly damaging to google.

I'm not going to bother reading a 4 year old article, and I don't understand what you're asking. We allow all Safe Browsing by default except the real-time binary checking (which we warn about) - i.e is we allow all local lists to update and be used, whatever it is they contain (and there are no privacy concerns here), but don't want to send google what is described in the `0412` description

1. b) auto-updating Firefox

I know how it works on windows, Linux not so much, depends on how it was "installed". On most/all? distros Firefox comes prebundled and in my VMs I tend to totally ignore that instance and I create new Firefox instances, one per version, and disable the auto-update (via the UI) so I have a test-suite

something like (from my notes: I am not super linux savvy, sorry)
```
- https://www.mozilla.org/en-US/firefox/all/
- download each tar.bz2 file
- make sure to get the 64bit version
- extract each one, rename the unpacked parent folder, e.g Firefox -> FF78ESR or FF78 or FF89
  ^ extract: right click, extract here

Alternatively, you can extract the archive from the command line:
cd ~/Downloads/
tar xjf firefox-89.0.tar.bz2

move them to opt
================

- open terminal
- cd ~/Downloads/
- sudo mv FF89 /opt/FF89

to remove old ones use
- sudo rm -r /opt/FF89

create links on desktop
=======================

- create new launchers on desktop
- use       /opt/FF89/firefox -no-remote -ProfileManager
- edit to   /opt/FF89/firefox -no-remote -p "ff89"
```

and this way I have one non-updating FF per release, for testing, which is not influenced by the OS/distro updates. I'm also pretty sure that you can update within the application in these: i.e check help>about, click update

For users, not my test suite, the main thing would be that you are notified of updates, and keep up to date (manually). Nothing wrong with auto-updating Firefox, so little changes between releases TBH (in terms of user.js tweaks) - it's your personal choice (the main thing is to not update in a timely fashion: IDK why so many people get upset over this notification, FFS).

**Comment by Thorin-Oakenpants** [comment]
2. delays ... AFAIK everything is pretty much speedy at my end (but then I've been using the user.js for 5 or 6 years, although I do use Nightly with no mods: can't say I notice any difference TBH)

- going forward/back shouldn't make a difference: it's all in memory which is faster than cache
- dns prefetching is basically non-existant: it is disabled by default for all Firefox users for HTTPS, and most of the web is HTTPS and the user.js explicitly uses HTTPS-Only Mode
   - and once something is in the DNS cache ... you get the idea
   - e.g. do a search, subsequent searches for that engine aren't going to requery the DNS
- make sure your bookmarks are for HTTPS where applicable, because loading one that is HTTP that upgrades probably causes lag
- link pre-fetching and connecting to the server in anticipation (`0601` and `0606`) are about links, not about the URLbar (but of course an already pre-fetched DNS/connection will speed shit up if you happen to type/load an URL that was pre-fetched etc), so I don't see that as an issue to worry about

What happens with your real Firefox vs the same setup in your VM? Is it a VM bottleneck? You said "i already tried using this user.js file in a VM and while i managed to make everything work the way" .. but have you tried it bare metal?

**Comment by Thorin-Oakenpants** [comment]
> my current browser settings+OS+combination of add-ons already makes it quit unique,so adding this user.js file wouldn't make it even more

The user.js looks at security first, then tracking/privacy .. fingerprinting is the last thing to worry about: and it is not guaranteed.

RFP (`4501`) will take the edge off a lot of metrics, and fool "naive" scripts as it randomizes canvas: most scripts are AFAanyoneKnows pretty dumb, and a lot of known scripts are already blocked (e.g. ETP fingerprinters list, uBO) and your first strategy is control what gets into your browser. That's changing, e.g. scripts are becoming first party and embedded in critical files: and that's where RFP comes into play: it's the last line of defense so to speak. Advanced scripts will have a field day to differing degrees, and RFP mitigates a lot of that - but ultimately it requires some more work and a large set of users (enforced, like Tor Browser), but for what it covers, along with say disabling webGL, you're going to be a lot better off than doing nothing - do nothing and you ARE unique

tl;dr: don't sweat it: use RFP and be aware of the three or so things that cause breakage
- canvas (you will see a canvas icon in the urlbar: allow a site exception for the session or permanently) - this does not compromise tracking you across all sites
- timing yank in games etc - play said games on another browser/profile
- shortcut keys with alt, shift can affect some sites like google docs
- some sites have blurry images (due to devicePixelRatio being spoofed)
- not breakage: it's behaving exactly as it is meant to: time is UTC0, just be aware when looking at times for events, or how this affects timestamps in web-based mail etc

That's about it for breakage/problems. RFP is a godsend, use it. Feel free to turn off the letterboxing (`4504`) - as far as you're concerned, it's not going to make much difference: naive scripts are already fooled, advanced ones wouldn't treat screen metrics as very stable (unless you are fullscreen or maximized) - the science of linking dynamic fingerprints is a whole other field

If that's it, feel free to close

**Comment by ginick** [comment]
thanks for clearing the Firefox update doubt.

quit a few settings are available under [SETUP-PERF],not sure which one will alleviate the lag.

as i was rereading instructions,i saw this: 
"non-RFP users may want to enable section 4600";
does this apply to the The main pref 4501,or everything RFP related?
i have ```user_pref("privacy.resistFingerprinting", true``` commented out and same for ```user_pref("privacy.resistFingerprinting.letterboxing", true)``` which is "independent of 4501".
the other settings are set to default.
so can i still enable section 4600?

**Comment by Thorin-Oakenpants** [comment]
`non-RFP` means `privacy.resistFingerprinting` = `false`. If you're not using RFP then it's up to you if you want to use anything in the 4600 section: personally, I wouldn't: it doesn't do anywhere near enough and is likely to cause issues: just get CanvasBlocker and use it to randomize your canvas (and nothing else unless you want to) and you'll fool naive scripts if they get through

**Comment by ginick** [comment]
Thanks Thorin,
i haven't tried it on bare metal yet,
I'm in the process of tweaking,adjusting and testing it first.
so maybe indeed this lag will disappear later.

i can't recall why i disabled RFP,i think it was to be able to perform search engine searches in the URL bar without it trying to translate it to an http:// address.

**Comment by rusty-snake** [comment]
> i can't recall why i disabled RFP,i think it was to be able to perform search engine searches in the URL bar without it trying to translate it to an http:// address.

That's
https://github.com/arkenfox/user.js/blob/a6d20eaf5b4d6fc306257cf56beb89595c6be173/user.js#L432-L439

**Comment by ginick** [comment]
you are right,i see i already took care of that preference,so it might have been something else.

anyhow i think that's it for now.

Thank you very much both for the assistance.

**Comment by Thorin-Oakenpants** [comment]
@ginick - rather than discard all the benefits of RFP, consider using an extension (or stylish) to dark theme **_ just those few websites_**. If you log into those sites (not DDG of course, which you can use without JS) then the FP isn't going to matter (if they are FPing), especially when you're not letting all sites know you modify web content colors

Some sites also use a cookie setting to remember your choices .. DDG used to (IDK about today), and ArsTechnica does - no account required. Github does too (but the setting is tied to your account, thus you need an account, doh). At the end of the day, as much as I love dark themed sites (`prefers-color-scheme`, see RFP returns as `light`), I decided I can live without it

**Comment by ginick** [comment]
would it be possible to still get a dark theme on supporting sites,
if i disable section 4600 (which I'm currently using),re-enable RFP, but still use ```user_pref("ui.systemUsesDarkTheme", 1); // [HIDDEN PREF]``` from section 4600 ?

**Comment by rusty-snake** [comment]
RFP overrides `ui.systemUsesDarkTheme`. But you can use extensions or settings of the site.

**Comment by Thorin-Oakenpants** [comment]
yeah, as Rusty points out, that pref is ignored for **web content** when RFP is on. I use `ui.systemUsesDarkTheme` to force the UI to be dark (and I use `system theme` because dark theme in windows 7 uses a white tabstrip due to a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1702480))

> if i disable section 4600 (which I'm currently using)

don't forget to reset them in about:config. Personally I don't think anyone should use section 4600, as without RFP you're already cooked and trying to cover a couple of metrics is a waste of time

> re-enable RFP

do it. live with it, and as you use sites that you frequent a lot that you feel must be dark, find a solution for getting it to use a dark theme. I don't use dark reader, and if that has a default don't-use with an allowlist, that could be perfect for you (as I said, it doesn't really compromise your FP if you log in, or it's only a few select sites)

**Comment by ginick** [comment]
Thank you,
I'll take all that into consideration.

**Comment by novocaiin** [comment]
> and I use `system theme` because dark theme in windows 7 uses a white tabstrip due to a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1702480)

Are you saying you're still using Windows 7?

**Comment by Thorin-Oakenpants** [comment]
yes. It's a very locked down win7 and a very secure network IMO, so I am not worried about it being EOL - 20% of FF users are on Win7 for what it's worth - so this is quite handy for my testing.

Not everyone has money and can afford shit. This machine is into it's 10 year (and still kicks ass - new graphics card last year - obviously not as performant on say encoding video, but I don't do that so it;s all good) - bought the pieces individually, built it myself for less than half the price of a store one - and it was damn near close to bleeding edge at the time and cost me two small arms and a leg (luckily I had some in the freezer). Win10 on here would cripple it beyond my ability to bear it

Now I have saved up and am about to buy a new rig and a laptop - I will probably make my rig Linux (for all my stuff) and my laptop (with touch and a a devicePixelRatio > 1 for testing) just be win10/11 for testing and so I can read shit out in the garden in the nude on sunny days

Will be a bit of a learning curve on Linux, but the computer spec upgrades will be something else - I'm thinking 128 core :)

**Comment by ginick** [comment]
Win7 was the best windows imo. clean and stable.
Win10/11 is great,but full of bloat, and the user ability to tweak and change things is slowly being taken away.

i moved to Linux around a year ago and it has indeed a learning curve and needs the user to be able/willing to do a lot of compromises. 
 
i recommend Manjaro which i am using.
